Sources/StreamChat/APIClient/RequestEncoder.swift

```diff
@@ -70,7 +70,7 @@ extension RequestEncoder {
 }
 
 /// The default implementation of `RequestEncoder`.
-struct DefaultRequestEncoder: RequestEncoder {
+class DefaultRequestEncoder: RequestEncoder {
```

Comment:
Why do we change it to a class?

Reply:
We need the reference semantics. Because not this is created in the init without keeping any reference to the encoder. From what I remember, all requests were failing if we kept them as structs.